server/models/docker_image.py

```diff
@@ -0,0 +1,279 @@
+
+from six import iteritems, string_types
+import hashlib
+import jsonschema
+
+
+class DockerImageError(Exception):
+    def __init__(self, message, image_name='None'):
+
+        self.message = message
+        # can be a string or list
+        self.imageName = image_name
+        Exception.__init__(self, message)
+
+    def __str__(self):
+        if isinstance(self.imageName, list):
+            return self.message + ' (image names ' \
+                                  '[' + ','.join(self.imageName)+']'
+        elif isinstance(self.imageName, string_types):
+            return self.message + ' (image name: ' + self.imageName+' )'
+        else:
+            return self.message
+
+
+class DockerImageNotFoundError(DockerImageError):
+    def __init__(self, message, image_name, locations=[]):
+        super(DockerImageNotFoundError, self).__init__(message, image_name)
+        # list of registries tried(local dockerhub etc )
+        self.locations = locations
+
+
+class DockerImageDataError(DockerImageError):
+    def __init__(self, message, image_name):
+        super(DockerImageDataError, self).__init__(message, image_name)
+
+
+class DockerImage():
+    """
+    Represents docker image and contains metadata on a specific image
+    """
+    # keys used by the dictionary that stores metadata on the image
+    imageName = 'docker_image_name'
+    imageHash = 'imagehash'
+    type = 'type'
+    xml = 'xml'
+    cli_dict = 'cli_list'
+    # structure of the dictionary to store meta data
+    # {
+    # imagehash:<hash of docker image name>
+    #     cli_list: {
+    #         cli_name: {
+    #                   type: < type >
+    #                   xml: < xml >
+    #
+    # }
+    # }
+    # docker_image_name: < name >
+    # }
+
+    def __init__(self, name):
+        try:
+            if isinstance(name, string_types):
+
+                imageKey = DockerImage.getHashKey(name)
+                self.data = {}
+                self.data[DockerImage.imageName] = name
+                self.data[DockerImage.cli_dict] = {}
+                self.data[DockerImage.imageHash] = imageKey
+                self.hash = imageKey
+                self.name = name
+                # TODO check/validate schema of dict
+            elif isinstance(name, dict):
+                jsonschema.validate(name, DockerImageStructure.ImageSchema)
+                self.data = name.copy()
+                self.name = self.data[DockerImage.imageName]
+                self.hash = DockerImage.getHashKey(self.name)
+            else:
+
+                raise DockerImageError('Image should be a string, or dict'
+                                       ' could not add the image',
+                                       'bad init val')
+        except Exception as err:
+                raise DockerImageError('Could not initialize instance'
+                                       ' of Docker Image \n'+err.__str__())
```

Comment:
In python the `__str__` method is called when you ask for a object's string representation. `str(err)` is preferred to `err.__str__()`.  Moreover, you can use printf-style formatting: `raise DockerImageError('Could not initialize instance of Docker Image: %s' % err)`, which internally will invoke the `__str__` method.